plugins/catalog/src/js/repositories/RepositoriesTab.js

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import { Observable } from "rxjs/Observable";
+
+// Recompose helps handle data down and event up as streams
+import {
+  createEventHandler,
+  componentFromStream
+} from "#PACKAGES/data-service/recomposeRx";
+
+// Our in-house solution to resolve graphql queries using streams
+import { graphqlObservable } from "#PACKAGES/data-service/graphqlObservable";
+
+// tools to make easier to create schemas and queries
+import { makeExecutableSchema } from "graphql-tools";
+import gql from "graphql-tag";
+
+// Streams we get our data from
+import {
+  liveFetchRepositories
+} from "#PLUGINS/catalog/src/js/repositories/repositoriesStream";
+import RepositoryList from "#SRC/js/structs/RepositoryList";
+
+// The graphql schema and resolvers for those streams;
+import { typeDefs, resolvers } from "./repositoriesModel";
+
+// UI components
+import RepositoriesTabUI from "./RepositoriesTabUI";
+
+// Using the data layer
+
+// 1. We first make an schema out of the resolvers and typeDefinitions
+const schema = makeExecutableSchema({
+  typeDefs,
+  resolvers
+});
+
+// 2. We declare which data our component needs in graphql (data down)
+const packageRepositoryQuery = gql`
+  query {
+    packageRepository(filter: $filter) {
+      name
+      uri
+    }
+  }
+`;
+
+// 3. We create an event handler as a stream to deal with search (events up)
+const { handler: onSearch, stream: searchEvent$ } = createEventHandler();
```

Comment:
Instead of using recompose I'd use `Subjects` as this would also allow us to specify how to handle events emitted before we subscribe to the "event" stream. 

```
const searchTerm$ = new BehaviorSubject("initial search term or `nothing`"); 

// ...

 <RepositoriesTabUI
        repositories={data.packageRepository}
        searchTerm={data.searchTerm}
        onSearch={searchTerm$.next}
      />

// ... 
```

Further more you can omit the `startWith` in case of a `BehaviorSubject` as it allows you to define an initial value.